sw/ground_segment/python/payload_decoder/payload_decoder.py

```diff
@@ -0,0 +1,167 @@
+
+        self.jpgdata = jpegheader + bytearray(self.thumb)
+        file_jpgdata = StringIO(self.jpgdata)
+        img = Image.open(file_jpgdata)
+        return img
+
+
+
+class PayloadDecoderFrame(wx.Frame):
+
+    def message_recv(self, ac_id, msg):
+        if msg.name == "PAYLOAD":
+            pld = msg.get_field(0).split(",")
+            b = []
+            for p in pld:
+                b.append(int(p))
+            self.thumb.add_payload(b)
+            self.data = b
+            sock.sendto(bytearray(b), (UDP_IP, UDP_PORT))
+            wx.CallAfter(self.update)
+
+    def update(self):
+        self.Refresh()
+
+    def OnSize(self, event):
+        self.w = event.GetSize()[0]
+        self.h = event.GetSize()[1]
+        self.Refresh()
+
+    def GetRGB(self, x, y, bpp):
+        # calculate some colour values for this sample based on x,y position
+        r = g = b = 0
+        if y < self.h/3:                           r = 255
+        if y >= self.h/3 and y <= 2*self.h/3:      g = 255
+        if y > 2*self.h/3:                         b = 255
+
+        if bpp == 4:
+            a = int(x * 255.0 / self.w)
+            return r, g, b, a
+        else:
+            return r, g, b
+
+    def MakeBitmapRGB(self, width, height):
+        # Make a bitmap using an array of RGB bytes
+        bpp = 3  # bytes per pixel
+        bytes = array.array('B', [0] * width*height*bpp)
+
+        for y in xrange(height):
+            for x in xrange(width):
+                offset = y*width*bpp + x*bpp
+                r,g,b = self.GetRGB(x, y, bpp)
+                bytes[offset + 0] = r
+                bytes[offset + 1] = g
+                bytes[offset + 2] = b
+
+        self.rgbBmp = wx.BitmapFromBuffer(width, height, bytes)
+
+
+    def OnPaint(self, e):
+        tdx = -5
+        tdy = -7
+
+        w = self.w
+        h = self.w
+
+        self.MakeBitmapRGB(w,h)
+
+        dc = wx.PaintDC(self)
+        brush = wx.Brush("white")
+        dc.SetBackground(brush)
+        dc.Clear()
+
+        # Background
+        dc.SetBrush(wx.Brush(wx.Colour(0,0,0), wx.TRANSPARENT))
+        #dc.DrawCircle(w/2,w/2,w/2-1)
+        font = wx.Font(11, wx.ROMAN, wx.BOLD, wx.NORMAL)
+        dc.SetFont(font)
+        #dc.DrawText("PAYLOAD",2,2)
+        dc.DrawText(str(self.data),2,2)
+
+        myPilImage = self.thumb.get_image()
+        myWxImage = wx.EmptyImage( myPilImage.size[0], myPilImage.size[1] )
+        myWxImage.SetData( myPilImage.convert( 'RGB' ).tostring() )
+        #print(myPilImage,myWxImage)
+        bitmap = wx.BitmapFromImage(myWxImage)
+        dc.DrawBitmap(bitmap, 30,30, True)
+
+        #c = wx.Colour(0,0,0)
+        #dc.SetBrush(wx.Brush(c, wx.SOLID))
+        #dc.DrawCircle(int(w/2),int(w/2),10)
+
+
+
+    def __init__(self):
+
+        self.thumb = ThumbNailFromPayload()
+        self.data = [];
+
+        self.w = WIDTH
+        self.h = WIDTH
+
+        wx.Frame.__init__(self, id=-1, parent=None, name=u'PayloadDecoder',
```

Comment:
As mentioned before, maybe this tool should be called "Thumbnails decoder" or equivalent, but something closer to the actual purpose of the program.